### DISCUSSION TITLE
Including documentation that explains how to include modules not officially supported.

### DISCUSSION THREAD

**Discussion by TheGuyWhoo** [DISTRHO/Cardinal#144] [Ideas]
Would this be an option? For those of us who don't have the coding know how who would like to add the FrozenWasteland and Stoermelder, etc. stuff that would be really cool.

**Comment by falkTX** [comment]
Technical skills are needed if you want to load non-supported modules, there is no way around it.
The Cardinal project does not have as a goal to facilitate that.

**Reply by TheGuyWhoo** [reply]
If I understand how it works correctly you'd only need a different plugin.cpp and makefile. Otherwise just copy and pasting the necessary code and following instructions for where to put it doesn't require much technical skill as well.

I do understand why it's not something you're interested in though.

**Reply by falkTX** [reply]
I mean you asked this before in #28 and I answered there.
Those are the same steps that apply today, nothing changed.

**Reply by TheGuyWhoo** [reply]
No, here I am asking about documentation that includes all the necessary code pertaining to specific modules, so people can download the source code for the modules themselves and just copy and paste the code from the documentation into plugin.cpp and and the makefile. I know where the code should go, I just don't actually know what any of it means. For example I managed to add Mojo modules by just following the template of Mog, but that doesn't work for other modules such FrozenWasteland and PdArray.

So, similar to the explanation you provided there, just pertaining to other modules that won't ever be included in Cardinal officially.

**Reply by falkTX** [reply]
how do you expect that to work though? it is required to change the makefiles and `plugins.cpp` in order to get more modules built into Cardinal.
the steps described on that other discussion are the documentation itself for this, there is no other way.

in order to have any code that you could in theory just copy&paste, someone still has to write it first. it doesn't appear out of nowhere..

**Reply by TheGuyWhoo** [reply]
The same way you you explained how to add cf, but instead you explain how to add FrozenWasteland and other modules that won't officially be included.

**Reply by falkTX** [reply]
that requires me to actually write the code to add them first.. which won't happen, why would I spend time on this?

**Reply by TheGuyWhoo** [reply]
To allow people a wider selection of modules that can do things that the officially supported plugins can't. It's just a suggestion.

**Reply by falkTX** [reply]
I dont think you understand the amount of time it takes to add new modules.
It is not just copy&paste, we need to fix any build issues first, sort out any extra build dependencies the projects might have, patch the file dialogs, check and fix symbol conflicts with existing modules that use similar names in their code, add in any binaries files that they use, and typically some testing/checking too.
Not to mention checking the licenses of their artwork/panels too.

No one is going to bother doing this for modules that are never going to be added in officially...

**Reply by TheGuyWhoo** [reply]
No I understand, I'm not you so I can't devine what you consider worth your time. Thus I ask.

**Reply by falkTX** [reply]
I can totally understand wanting some of these modules.
But honestly time is better spent talking to them to resolve any possible conflicts, and try to have them do that work themselves as it would not be complicated for them.

Spending half a day trying to get module X or Y to work on Cardinal and knowing fully well that license-wise they are not compatible.. it is just not a good way to spend the time.
Licenses are not something to take lightly. I went above and beyond to make sure everything that is included in Cardinal is legal, authorized, and good for the long-term of the project. I do not wish to compromise that.

**Reply by TheGuyWhoo** [reply]
Fair enough
